src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -267,44 +290,74 @@ def onMouseClickOnImage(self, x: float, y: float):
             radial_value + self._delta_radial_over_2,
         )
 
-    def getColorLastCurve(self):
-        curves = self._integrated_plot_widget.getAllCurves()
-        if curves:
-            return self._integrated_plot_widget.getAllCurves()[-1].getColor()
+    def getColorCurve(self, legend: str):
+        curve = self._integrated_plot_widget.getCurve(legend=legend)
+        if curve:
+            return curve.getColor()
         else:
             return
 
+    def getColorAvailable(self, legend: str):
+        if self._integrated_plot_widget.getCurve(legend=f"INTEGRATE{legend}"):
+            color = self._integrated_plot_widget.getCurve(legend=f"INTEGRATE{legend}").getColor()
+        else:
+            index_color = 0
+            color = rgba(self._integrated_plot_widget.getDefaultColors()[index_color])
+            while color in [self._integrated_plot_widget.getCurve(legend=legend).getColor() for legend in self._integrated_plot_widget]:
```

Comment:
This while loop becomes infinite if there are more than 10 curves pinned.

I think we can instead use the method `_getColorAndStyle` from the `PlotWidget`: https://github.com/silx-kit/silx/blob/755fa6d3cbe35e1078f65557f6e9e8352b826da5/src/silx/gui/plot/PlotWidget.py#L3086